frontend/src/visualization/Graph.tsx

```diff
@@ -76,7 +77,7 @@ function Graph(props: GraphProps): JSX.Element {
   // The size of the container that is used to measure the available space for the graph.
   const containerSize = useSize(sizeMeasureContainerRef);
 
-  const graphDataStore = useService(QueryResultStore);
+  const graphDataStore = useService<QueryResultStore>(QueryResultStore);
```

Comment:
TS can infer the type automatically. It also does not change the code at runtime.
```suggestion
  const graphDataStore = useService(QueryResultStore);
```